python/ray/tests/test_runtime_env_working_dir.py

```diff
@@ -606,6 +607,59 @@ class B:
         B.options(runtime_env={"working_dir": "."})
 
 
+@pytest.mark.skipif(sys.platform == "win32", reason="Fail to create temp dir.")
+@pytest.mark.parametrize("disable_working_dir_gc", [True, False])
+def test_id_named_working_dir(tmp_working_dir, disable_working_dir_gc, shutdown_only):
+    """Tests the independent id named working directory for echo worker."""
+    if disable_working_dir_gc:
+        os.environ["RAY_RUNTIME_ENV_DISABLE_WORKING_DIR_GC"] = "true"
+    else:
+        os.environ["RAY_RUNTIME_ENV_DISABLE_WORKING_DIR_GC"] = "false"
+    ray.init(runtime_env={"working_dir": tmp_working_dir})
+
+    @ray.remote
+    class A:
+        def test_import(self):
+            import test_module
+
+            return test_module.one()
+
+        def get_cwd(self):
+            return os.getcwd()
+
+        def write_file(self, file_name):
+            with open(file_name, "w") as file:
+                file.write("working_dir_test")
+            return True
+
+        def check_file(self, file_name):
+            return os.path.exists(file_name)
+
+    a = A.remote()
+    assert ray.get(a.test_import.remote()) == 1
+    actor_cwd = ray.get(a.get_cwd.remote())
+    assert "working_dirs" in actor_cwd
+    assert os.path.exists(actor_cwd) and os.path.isdir(actor_cwd)
+    ray.kill(a, no_restart=True)
+
+    def _check_path_does_not_exist(path):
+        return not os.path.exists(path)
+
+    if not disable_working_dir_gc:
+        wait_for_condition(_check_path_does_not_exist, path=actor_cwd, timeout=3)
+    else:
+        time.sleep(5)
```

Comment:
The `sleep` will increase the time consuming of unit test and case it flaky when node overload, please optimize it.